### DISCUSSION TITLE
Enabling combobox when fill the previous one

### DISCUSSION THREAD

**Discussion by MrDerekib** [TomSchimansky/CustomTkinter#2619] [General]
I've done an internal tool with tkinter and I'm migrating to CustomTkinter for best looking and it's a think I can solve by my own. Will appreciate some help.

The "app" fills an excel with data, nothing fancy. 

I have some CTkComboBox in it, the first one is enabled (state= "normal") and the other ones are disabled (state="disabled"). 
When You pick an option of the first combobox, the next will be enabled an the rest remains disabled... you pick some value on the 2nd and the third will be enabled...

I have done this feature with tkinter but i can't with CustomTkinter. 

Any idea / method to solve it? 

I've have success with a button in CustomTkinter but it's not the finest experience

**Comment by dipeshSam** [comment]
@MrDerekib Could you please have sample code (either in Tkinter or CustomTkinter) for your requirements?

Further more information about your desired result would be helpful. Like the following:
- Which geometry manager are you looking for; `grid`, `pack` or `place`?
- In which order you want to operate them; in a row, column, or it is random.

Also, in your description:
> When You pick an option of the first combobox, the next will be enabled an the rest remains disabled...

After selection, the next one gets enabled, but what about the first one? Does the first one also go disabled or it remains enabled to change the selected value again?

Regards.

**Reply by MrDerekib** [reply]
> Which geometry manager are you looking for; grid, pack or place?

I don't know it's a simple form style app... I've tried with grid in my app.

> In which order you want to operate them; in a row, column, or it is random.

In a column, It's like repair1, repair2, repair3... The first is enabled by default (required field), the others disabled. Only will enable repair 2 if the repair1 is with an item selected and so on...

Sample code with a button, that's what I dont want to do if I can (I want to do that but with automatic enabling the next element when fill with data a combobox):

```python
import customtkinter as ctk

root = ctk.CTk()
root.title("Prueba de Combobox Secuenciales")
root.geometry("400x300")

datos_combobox1 = ["Opción 1", "Opción 2", "Opción 3"]
datos_combobox2 = ["Opción A", "Opción B", "Opción C"]
datos_combobox3 = ["Elemento X", "Elemento Y", "Elemento Z"]

combobox1 = ctk.CTkComboBox(root, values=datos_combobox1)
combobox1.grid(row=0, column=0, padx=20, pady=20)

combobox2 = ctk.CTkComboBox(root, values=datos_combobox2, state="disabled")
combobox2.grid(row=1, column=0, padx=20, pady=20)

combobox3 = ctk.CTkComboBox(root, values=datos_combobox3, state="disabled")
combobox3.grid(row=2, column=0, padx=20, pady=20)

def habilitar_siguiente():
    if combobox1.get():  # Si hay una selección en combobox1
        combobox2.configure(state="normal")
    if combobox2.get():  # Si hay una selección en combobox2
        combobox3.configure(state="normal")

btn_habilitar = ctk.CTkButton(root, text="Habilitar Siguiente", command=habilitar_siguiente)
btn_habilitar.grid(row=3, column=0, padx=20, pady=20)

root.mainloop()
``

**Comment by dipeshSam** [comment]
@MrDerekib  Here is sample implementation of the facility you want:
``` Python
from tkinter import Event
from customtkinter import CTk, CTkFrame, CTkComboBox
from typing import Callable, Any

class CombosFrame(CTkFrame):
    def __init__(self, master: CTk, callback: Callable[[str, tuple[int, int]], Any] = None, *args, **kwargs):
        super().__init__(master, *args, **kwargs)

        self.callback = callback
        self._active: CTkComboBox = None        
        self.items: list[CTkComboBox] = []
        
        
    def _on_click(self, event: Event):
        self._active = event.widget.master


    def _on_select(self, value: str):
        self._active_next()
        if self.callback:
            grid_info = self._active.grid_info()
            self.callback(value, (grid_info["row"], grid_info["column"]))
    
    def _active_next(self):
        index = self.items.index(self._active) + 1
        combo: CTkComboBox = self.items[index if index < len(self.items) else index - 1]
        combo.configure(state="normal")
        

    def fill(self, items: list[list[CTkComboBox]]):
        """Takes list of columns of CTkComboboxes."""
        for col, combo_cols in enumerate(items):
            for row, combobox in enumerate(combo_cols):
                combobox.grid(row=row, column=col, padx=10, pady=10)
                combobox._canvas.bind("<Button-1>", self._on_click)
                combobox.configure(command=self._on_select)
                self.items.append(combobox)
            
        # Enabling first one
        items[0][0].configure(state="normal")
        
            
            
if __name__ == "__main__":
    def on_select(*data):
        print(data)
    
    app = CTk()
    
    frame = CombosFrame(app, callback=on_select)
    frame.pack(fill="both", expand=True, padx=20, pady=40)

    sample_values = ("Sample1", "Sample2", "Sample3")
    combos = [[CTkComboBox(frame, values=sample_values, state="disabled") for _ in range(5)] for _ in range(3)]

    frame.fill(combos)

    app.mainloop()        
```
**Output:**
![consecutive comboboxes in CTk](https://github.com/user-attachments/assets/7e0992f5-357b-4e5a-b64b-6e123985077c)



As in your example code, only one column is used, to do so, replace `range(3)` with `range(1)`. Let me know what you expect more from the provided code.

Regards.
